### ISSUE TITLE
Missing import isn't detected

### ISSUE THREAD

**Issue by willbar** [Microsoft/TypeScript#11506]
**TypeScript Version:**  2.0.3

**Code**

```ts
import express = require('express');

export const router = express.Router();

router.post('/witchange', (req: express.Request, res: express.Response, next: express.RequestHandler) => {
  const promises = [];
  Q.all(promises).then(
    () => res.status(200).end('success'),
    (error) => res.status(500).end('internal server error')
  );
});

```

**Expected behavior:**
Error that Q is not defined.

**Actual behavior:** 
No error. If you can't repro this from this snippet I can share the full repository offline.

**Comment by mhegazy** [comment]
I can not repro this locally; where is `Q` defined for you? in your editor, F12 on `Q`, where does it take you?

**Comment by willbar** [comment]
It takes me to typings/globals/q/index.d.ts (where it's getting that from I'm not sure because there's no import of Q in the file) but at runtime I get:

```
Exception Message
Q is not defined
```

**Comment by willbar** [comment]
Interestingly, it's taking me to the Q namespace definition, so I would expect to get a compile error that Q.all is not a function.

**Comment by mhegazy** [comment]
looks like `Q` is defined somewhere. either remove the file, exclude it, or remove `/// <reference` to it.

**Comment by mhegazy** [comment]
> Interestingly, it's taking me to the Q namespace definition, so I would expect to get a compile error that Q.all is not a function.

if you hit F12 on `Q.all`, does it take you to a function?

**Comment by willbar** [comment]
Yes, which makes sense because there is a namespace level function defined in the .d.ts. Doesn't a namespace need to be imported into a file to be able to reference anything from it? It is failing at runtime but succeeding at compile time which feels like a bad combination.

**Comment by mhegazy** [comment]
No. a namespace is just a global object. it is always in your scope (unless shadowed). modules are the one that needs to be imported.

You can find more documentation in http://www.typescriptlang.org/docs/handbook/namespaces-and-modules.html

**Comment by willbar** [comment]
The documentation makes sense. The piece I'm missing is why does this fail at runtime? The value Typescript delivers me is detecting issues at compile time that would cause runtime failures, in this case the code is compiling successfully (per design it seems) but then fails at runtime. The runtime in this case being NodeJS 6.7.0.

**Comment by mhegazy** [comment]
The compiler uses the declaration files as a source of information about what `Q` looks like. the declaration file says `Q` is a global variable that always exists, just like `console`, `window`, etc.. so using it in such contexts is allowed.

Obviously this is not correct, as you get an error at rune time. so the declaration file can be updated to reflect that. Looks like the declaration has been updated to reflect this in `@types/q`. In https://github.com/DefinitelyTyped/DefinitelyTyped/blob/types-2.0/q/index.d.ts (which is what is published on `@types/q`), it is declared as a [UMD module](http://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html#umd), this should give you an error if you use it as a global like `"Identifier `Q` must be used as a module"`.

**Comment by willbar** [comment]
That does it, thanks!
